components/raftstore/src/store/worker/pd.rs

```diff
@@ -1346,6 +1347,26 @@ where
                             warn!("fail to schedule control grpc task"; "err" => ?e);
                         }
                     }
+                    // NodeState for this store.
+                    {
+                        let state = (|| {
+                            #[cfg(feature = "failpoints")]
+                            fail_point!("manually_set_store_offline", |_| {
+                                metapb::NodeState::Removing
+                            });
+                            resp.get_state()
+                        })();
+                        match state {
+                            metapb::NodeState::Removing | metapb::NodeState::Removed => {
+                                let is_offlined = snap_mgr.is_offlined();
+                                if !is_offlined {
+                                    snap_mgr.set_offline(true);
+                                    info!("store is offlined by pd");
+                                }
+                            }
+                            _ => {}
```

Comment:
what if offline is halted, then the state is back to normal

Reply:
Missed it, thx for reminding.